xy/convex_hull.go

```diff
@@ -237,9 +237,21 @@ func (calc *convexHullCalculator) computeOctRing(inputPts []float64) []float64 {
 	}
 
 	copyTo += stride
-	octPts = octPts[0 : copyTo+stride]
+
+	// if ring already closed, just return
+	if internal.Equal(octPts, copyTo-stride, octPts, 0) {
+		return octPts
+	}
 
 	// close ring
+	// make sure there is enough capacity
+	if len(octPts) < copyTo+stride {
+		t := make([]float64, copyTo+stride)
+		copy(t, octPts)
+		octPts = t
+	}
```

Comment:
I think what this PR is doing here is extending `octPts` with zeros, which is probably not quite right.

Reply:
you're probably right. I didn't really understand the problem, I just tried to inspect the data to guess what should happen.

Neither way, the test case in this PR is valid: without the fix, the test panics